File: lib/rest/yampi/mappers/yampi_collection_mapper.dart
import 'package:sertton/core/catalog/dtos/collection_dto.dart';
import 'package:sertton/rest/types/json.dart';

class YampiCollectionMapper {
  static CollectionDto toDto(Json json) {
    return CollectionDto(id: json['id'].toString(), name: json['name'] ?? '');
  }

  static List<CollectionDto> toDtoList(Json json) {
    final data = (json['data'] as List).cast<Json>();
    return data.map(toDto).toList();
  }
}

File: sertton-react-native-app/src/rest/yampi/mappers/yampi-collection-mapper.ts
import type { Collection } from "@/core/catalog/entities"
import type { YampiCollection } from "@/rest/yampi/types"

export const YampiCollectionMapper = () => ({
  toDomain(input: YampiCollection): Collection {
    return { id: String(input.id), name: input.name }
  },
})
